packages/sdk/src/services/client-task-queue-service.ts

```diff
@@ -64,6 +83,99 @@ export class ClientTaskQueueService {
       in_progress: inProgressTasks.client_tasks.length,
     };
   }
+
+  public async executeClientTasks(
+    listOfClientTasks: ClientTask[],
+    authData: AuthData
+  ): Promise<{ completedTasks: ClientTask[]; failedTasks: ClientTask[] }> {
+    const remainingClientTasks: ClientTask[] = [];
+    const itemUpdateSharesTasks: ClientTask[] = [];
+    for (const task of listOfClientTasks) {
+      switch (task.work_type) {
+        case 'update_item_shares':
+          itemUpdateSharesTasks.push(task);
+          break;
+        default:
+          remainingClientTasks.push(task);
+          break;
+      }
+    }
+    if (remainingClientTasks.length) {
+      throw new Error(
+        `Do not know how to execute ClientTask of type ${remainingClientTasks[0].work_type}`
+      );
+    }
+    const [updateSharesTasksResult]: Array<{
+      completedTasks: ClientTask[];
+      failedTasks: IFailedClientTask[];
+    }> = await Promise.all([this.updateSharesClientTasks(itemUpdateSharesTasks, authData)]);
```

Comment:
`Promise.all` not required here - `updateSharesClientTasks` returns a single promise.